### PR TITLE
Issues/75

### LINKED ISSUE #75: Bi-directional syncing not working on two devices
Now I don't know if I'm an idiot, and totally misunderstanding the point of this program, but I can't get bi-directional syncing to work on two devices. Example being I play game on computer A, save syncs to One drive, try to play game on computer b with same directory setup, and I get an error asking me if the log files exist (they do). I'll delete the lock files to make new files, and then it sinks again but it doesn't download the new data anymore. If I sync computer b data, then try to resync on computer A again, same issue. Don't really know what I'm doing wrong, unless I've misunderstood what this program is supposed to be for.

### LINKED ISSUE #81: Plugin incorrectly identifies some games as the ones with Steam Cloud and doesn't perform cloud sync
POP WW doesn't support Steam Cloud but the plugin claims that it does.

### LINKED ISSUE #92: Suggestion: Allow users to specify a different destination subfolder for each sync path
Similarly to #86 I also got frustrated with the long paths for save games - especially when the path does not have the game name inside but some kind of UUID and it's hard to keep track of which game is which. However I had a different idea in mind, why not just let users provide a specific subfolder name for each sync path? There could be a button next to the sync path delete button that let you specify a subfolder name. For example in my Dropbox, I might have /decky-cloud-save/GameName1 and /decky-cloud-save/GameName2 but the actual paths that sync to those destinations could still be things like /run/media/mmcblk0p1/some/long/path/with/uuid/savedir

### LINKED ISSUE #96: Bisync error in 1.4.1. - Critical error: "chtimes /home operation not permitted" when resync
Like in the title. When bisync and then resync these are the logs.

### LINKED ISSUE #97: Sync on start triggered on a Steam Cloud game
Tested with the free demo of this game https://store.steampowered.com/app/2401970/Ace_Attorney_Investigations_Collection/

This demo supports Steam Cloud, but still triggers the sync.

### PR DESCRIPTION
**Pull request by GedasFX**
Fixes #97 
Fixes #96 
Fixes #81 
Fixes #75 
Resolves? #45 
Resolves #92

### CODE REVIEW

**Comment by GedasFX** [comment]
@AkazaRenn How's this?

File: py_modules/plugin_config.py [hunk 1]
```diff
@@ -110,3 +110,5 @@ def migrate():
         set_config("toast_auto_sync", "true")
     if not any(e[0] == "additional_sync_args" for e in config):
         set_config("additional_sync_args", "")
+    if not any(e[0] == "sync_root" for e in config):
+        set_config("sync_root", "/")
```
Comment (AkazaRenn) [inline py_modules/plugin_config.py:114]: I suggest we don't set the variable in the config by default, so no one knows its existence before reading the document

Comment (GedasFX) [reply py_modules/plugin_config.py:114]: The reason why I set it by default is so users can see what needs to be changed, easier than explaining to add a new line. Pick your poison type of deal 🤷

**Comment by GedasFX** [comment]
Fixes #97 
Fixes #81 
Fixes #75 
Implements #45 
Implements #92

File: py_modules/plugin_config.py (hunk 1 above)
Comment (AkazaRenn) [reply py_modules/plugin_config.py:114]: Makes sense... Ok we'll just make them responsible for modifying the config file

**Review by AkazaRenn — APPROVED** [review]

**Comment by ajburley** [comment]
Thanks a lot, the solution sounds great! I guess it also resolves #86

**Comment by AkazaRenn** [comment]
Sorry just found a problem, once we've changed the sync root, the sync path filter is also changed to get the leading path removed. This is not a problem, but adding new paths on the GUI will be, we need to programmatically do that when adding paths.

**Comment by GedasFX** [comment]
@AkazaRenn 
image
I guess we can dynamically add the trailing / if needed. Didn't think of it at the time

**Pull request merged**